src/components/performances/PerformanceForm.jsx

```diff
@@ -19,20 +20,28 @@ class PerformanceForm extends React.Component {
     const { currentValues } = props;
 
     this.state = {
+      disabled: true,
       academicLevel: currentValues.academicLevel || '',
       choreographers: currentValues.choreographers || '',
       competitionLevel: currentValues.competitionLevel || '',
       danceEntry: currentValues.danceEntry,
       danceStyle: currentValues.danceStyle || '',
       danceTitle: currentValues.danceTitle || '',
-      disabled: false,
       options: {},
       performers: currentValues.performers || '',
       school: currentValues.school || '',
       size: currentValues.size || 0
     };
   }
 
+  // disable save button if not all input fields are filled
+  static getDerivedStateFromProps(props, state) {
```

Comment:
I'll circle back on this in person but I'm not sure if `getDerivedStateFromProps` is the best place to do this. We're actually not deriving the state from the props but instead from the state only.

Reply:
Keeping based on existing precedence, moving this to componentDidUpdate will cause double renders.